### ISSUE TITLE
Load BrokerClient Implementations Lazily

### ISSUE THREAD

**Issue by DiCanio** [num-codex/codex-feasibility-backend#12]
We should only load `BrokerClient` implementations that are actually required for running the backend. Since we can only choose one active path at a time, the other implementations do not have to be loaded.

This has 2 benefits:
- it saves us some space since less classes get loaded
- possible value realizations during startup get deferred

**Why is the latter benefit important?**

The different `BrokerClient` implementations may require some values to be provided via environment variables. However, Spring will complain about them not being set even if the path they are associated with is not chosen to be active. Deferred instantiation is a legitimate way to avoid this since values are only really read if necessary.

TL/DR:
Provides a legitimate way of only requiring environment variables of a chosen path.
